File: test_servers/test_server_spring/src/main/java/project/utils/classes/ResultType.java
package project.utils.classes;

public class ResultType<T> {
    private String result;
    private String type;
    private T attributes;
    public String getResult() {
        return result;
    }
    public void setResult(String result) {
        this.result = result;
    }
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }
    public T getAttributes() {
        return attributes;
    }
    public void setAttributes(T attributes) {
        this.attributes = attributes;
    }
    public ResultType() {
    }
    public ResultType(String result, String type, T attributes) {
        this.setAttributes(attributes);
        this.setResult(result);
        this.setType(type);
    }
}

File: test_servers/express-app/ResultType.ts
export class ResultType<T>{
    private result!: string;
    private type!: string;
    private attributes!: T;

    /**
     * Getter $result
     * @return {string}
     */
	public get $result(): string {
		return this.result;
	}

    /**
     * Setter $result
     * @param {string} value
     */
	public set $result(value: string) {
		this.result = value;
	}

    /**
     * Getter $type
     * @return {string}
     */
	public get $type(): string {
		return this.type;
	}

    /**
     * Setter $type
     * @param {string} value
     */
	public set $type(value: string) {
		this.type = value;
	}

    /**
     * Getter $attributes
     * @return {T}
     */
	public get $attributes(): T {
		return this.attributes;
	}

    /**
     * Setter $attributes
     * @param {T} value
     */
	public set $attributes(value: T) {
		this.attributes = value;
	}

	constructor(result: string, type: string, attributes : T) {
        this.$attributes = attributes;
        this.$result = result;
        this.$type = type;
	}

}